javascript/javascript1/week4/sentiment.js

```diff
@@ -0,0 +1,59 @@
+function getWordsFromText (text, wordsList) {
+   const words = [];
+   for (let i = 0; i < text.length; i++ ){
+      if(wordsList.includes(text[i])) {
+         words.push(text[i])
+      } 
+   }
+   return words
+}
+
+function getSentimentScore(text, arrPositiveWords, arrNegativeWords) {
+   const arrText = text.split(" ");
```

Comment:
`split(" ")` presents a problem if a word has comma exactly after it, which is the case when you list characteristics as "amazing, fantastic, disappointing". Those words with comma, will not be matched with the `includes` method. To prevent that issue, you can avoid creating an `arrText` and simple loop through the negative and positive lists and search for match:
```
for (let i = 0; i < listPositiveWords.length; i++) {
    if (sentence.match(listPositiveWords[i])) {
      positiveScoreWords.push(listPositiveWords[i]);
      score++;
    }
  }
```